retrieval_service/datastore/providers/firestore.py

```diff
@@ -35,47 +38,111 @@ class Config(BaseModel, datastore.AbstractConfig):
+                    "composite",
+                    "delete",
+                    index,
+                    "--database=(default)",
+                    "--quiet",  # Added to suppress delete warning
+                ]
+
+                delete_vector_index_process = await asyncio.create_subprocess_exec(
+                    *delete_vector_index,
+                )
+                delete_tasks.append(
+                    asyncio.create_task(delete_vector_index_process.wait())
+                )
+        await asyncio.gather(*delete_tasks)
+
     async def initialize_data(
         self,
         airports: list[models.Airport],
         amenities: list[models.Amenity],
         flights: list[models.Flight],
         policies: list[models.Policy],
     ) -> None:
-        async def delete_collections(collection_list: list[AsyncCollectionReference]):
-            # Checks if colelction exists and deletes all documents
-            delete_tasks = []
-            for collection_ref in collection_list:
-                collection_exists = collection_ref.limit(1).stream()
-                if not collection_exists:
-                    continue
-
-                docs = collection_ref.stream()
-                async for doc in docs:
-                    delete_tasks.append(asyncio.create_task(doc.reference.delete()))
-            asyncio.gather(*delete_tasks)
-
         # Check if the collections already exist; if so, delete collections
         airports_ref = self.__client.collection("airports")
         amenities_ref = self.__client.collection("amenities")
         flights_ref = self.__client.collection("flights")
         policies_ref = self.__client.collection("policies")
-        await delete_collections(
+        await self.delete_collections(
             [airports_ref, amenities_ref, flights_ref, policies_ref]
         )
 
+        # List indexes and retrieve name field (file-path)
+        list_vector_index = [
+            "gcloud",
+            "alpha",
+            "firestore",
+            "indexes",
+            "composite",
+            "list",
+            "--database=(default)",
+            "--format=value(name)",  # prints name field
+        ]
+
+        list_vector_index_process = await asyncio.create_subprocess_exec(
+            *list_vector_index,
+            stdout=asyncio.subprocess.PIPE,
+        )
+
+        # Capture output and ignore stderr
+        stdout, __ = await list_vector_index_process.communicate()
+
+        # Decode and format output
+        indexes = stdout.decode().strip().split("\n")
+
+        # Check if the indexes already exist; if so, delete indexes
+        # Extract collection and index-id from file path
+        # Assign the index-id to the corresponding collection
+        if indexes != [""]:
+            collections = {"amenities": "", "policies": ""}
+            for line in indexes:
+                collection, index_id = line.split("/")[-3], line.split("/")[-1]
```

Comment:
you should consolidate all of the string parsing in a single place. Doing this will give you more flexibility later to change how you implement this, without this part needing to be rewritten